Library/Homebrew/exceptions.rb

```diff
@@ -87,21 +87,31 @@ def initialize(name)
     super()
 
     @name = name
+    @without_api = Homebrew::EnvConfig.no_install_from_api?
+    @auto_without_api = Homebrew::EnvConfig.automatically_set_no_install_from_api?
   end
 
   sig { returns(String) }
   def did_you_mean
     require "formula"
 
-    similar_formula_names = Formula.fuzzy_search(name)
+    similar_formula_names = if @without_api
+      Homebrew.without_api { Formula.fuzzy_search(name) }
```

Comment:
Feels like this repeated pattern could be a made a bit nicer with a `yield` and something like a `Homebrew.without_api_if_needed(@without_api) { ... }` block?